cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java

```diff
@@ -764,11 +844,79 @@ private void createBanners(Collection<Banner> banners) {
             break;
           }
         }
+      }
+
+      boolean targetEvents = false;
+      if (banner.getEventFilters() != null && banner.getEventFilters().size() > 0) {
+        isTargetEvent = true;
+        for (BannerTargetEvent bannerTargetEvent : banner.getEventFilters()) {
+          boolean targetConditionTrue;
+          String event = bannerTargetEvent.getEvent();
+          String property = bannerTargetEvent.getProperty();
+          String relationString = bannerTargetEvent.getRelation();
+          String value = bannerTargetEvent.getValue();
+          String fromValue = bannerTargetEvent.getFromValue();
+          String toValue = bannerTargetEvent.getToValue();
+
+          if (event == null || event.isEmpty() ||
+                  property == null || property.isEmpty() || Double.parseDouble(property) < 0 || Double.parseDouble(property) % 1 != 0 ||
+                  relationString == null || relationString.isEmpty() ||
+                  (relationString.equalsIgnoreCase("between") &&
+                          (fromValue == null || fromValue.isEmpty() || Double.parseDouble(fromValue) < 0 || Double.parseDouble(fromValue) % 1 != 0 ||
+                                  toValue == null || toValue.isEmpty() || Double.parseDouble(toValue) < 0 || Double.parseDouble(toValue) % 1 != 0)) ||
+                  (!relationString.equalsIgnoreCase("between") && (value == null || value.isEmpty() || Double.parseDouble(value) < 0 || Double.parseDouble(value) % 1 != 0))) {
+            Logger.d(TAG, "Skipping Target in banner " + banner.getId() + "because: Target values are not valid");
+            continue;
+          }
+
+          ArrayList<TableBannerTrackEvent> bannerTrackEvents = (ArrayList<TableBannerTrackEvent>) DatabaseClient.getInstance(CleverPush.context).
+                  getAppDatabase()
+                  .trackEventDao()
+                  .getBannerTrackEvent(banner.getId(), event, property, value != null ? value : "", String.valueOf(CheckFilterRelation.fromString(relationString)), fromValue != null ? fromValue : "", toValue != null ? toValue : "");
```

Comment:
String.valueOf(CheckFilterRelation.fromString(relationString) put into an extra vairable, and use multiple lines here please